src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -280,7 +280,7 @@ private string GetVersion(string assetId)
             return VersionIdentifier.GetVersion(assetId);
         }
 
-        private (List<BuildData>, List<SigningInformation>, ManifestBuildData) GetBuildManifestsMetadata(
+        internal (List<BuildData>, List<SigningInformation>, ManifestBuildData) GetBuildManifestsMetadata(
```

Comment:
We'll need to change Directory (and FileStream) to use the Proxy pattern so that we can mock this out. We shouldn't have to rely on the existence of files or a specific directory set up (with manifestsFolderPath) to run our tests. (This goes hand-in-hand with my other comment regarding the making the smaller XML files strings so that we can use those results in mocks).

Reply:
Do you think it's really worth the time to mock FileStream for this? Or basically pretend Directory & FileStream aren't there and jump straight to the parsed XML? This is probably my functional/scenario test writing experience talking, but every removed layer within the product code creates more uncertainty that everything is covered, in my mind anyway.

Reply:
For unit testing it is definitely worth mocking dependencies. Removing the need to provide "physical" file assets to the code base for testing will reduce bloat in a repo (e.g. adding MSI files in some of our repos has caused them to increase in size and require more time to download). The class' unit tests aren't responsible for testing third party dependencies, which is what Directory and FileStream are in this case. All that should matter is that when those results are provided to the class from the methods called on Directory and FileStream, that your feature code handles it properly. 

This is not to say that scenario tests aren't important, but that a lot of bugs can be found faster through unit testing, as there shouldn't be any sort of environment requirements (including requiring certain files to be in a certain directory structure) for running unit tests.

Reply:
If it can be factored out a bit so the file operations are easy to just skip with the tests, that works too.  (If, for example, all the files are parsed, and then the contents passed to some other method, you can just test the second method and ignore all the file stuff).  That's not always possible though.

Reply:
There is an IFileProvider thingy in Microsoft.Extensions.FileProviders.Abstractions that we could use to factor these out. Its what is used for all the asp.net core file stuff.

Reply:
It is read-only though, so if we need read/write it won't work.

Reply:
Refactored the product method to avoid needing these mocks.